### DISCUSSION TITLE
Why was the metric endpoint removed?

### DISCUSSION THREAD

**Discussion by posquit0** [argoproj/argo-workflows#10289] [Q&A] [answered]
I asked in the following issue thread, but I've not received an answer.
- https://github.com/argoproj/argo-workflows/pull/9399

@terrytangyuan @alexec
In https://github.com/argoproj/argo-workflows/issues/8441 I can't see why maintainers decided to remove the metrics service. What service should I use to collect Prometheus metrics after this service is removed?

**Comment by sarabala1979** [comment, marked as answer]
@posquit0 The controller will currently support HA. The standby controller will not open the metrics port until it becomes a leader. If you have a service for controllers, it will get 404 from standby.

Are you using Prometheus? you can use PodMonitors for collection metrics.

OR
if you are not enabling HA, you can create a service

**Reply by posquit0** [reply]
Thank you for the answer!
